STIPS/STIP-009.md

```diff
@@ -0,0 +1,743 @@
+
+### StreamingFeeExtension.accrueFeeAndDistribute()
+
+![StreamingFeeExtension](../assets/stip-009/image5.png "")
+
+An interested party wants to accrue streaming fees and distribute them to the `owner` and `methodologist`.
+
+1. The interested party calls accrueFeeAndDistribute() on the StreamingFeeExtension
+2. Fees are accrued to the DelegatedManager
+3. Fees are distributed to the `owner` and `methodologist`
+
+
+## Checkpoint 2
+
+**Reviewer**:
+
+Reviewer: []
+
+## Specification
+
+### ManagerFactory
+
+#### Events
+
+##### DelegatedManagerDeployed
+
+| Type  | Name  | Description   |
+|------ |------ |-------------  |
+|address|setToken|Address of SetToken to be managed|
+|address|deployer|Address of the deployer|
+|address|owner|Address of the owner|
+|address|manager|Address of the DelegatedManager|
+
+
+##### DelegatedManagerInitialized
+
+| Type  | Name  | Description   |
+|------ |------ |-------------  |
+|address|setToken|Address of SetToken to be managed|
+|address|deployer|Address of the deployer|
+|address|owner|Address of the owner|
+|address|manager|Address of the DelegatedManager|
+
+
+#### Structs
+
+##### InitializeParams
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|address|deployer|Address of the deployer|
+|address|owner|Address of the owner|
+|address|manager|Address of the DelegatedManager|
+|bool|isPending|Bool if manager in pending state|
+
+#### Public Variables
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|address|factory|Address of Set Token factory|
+|mapping(address => InitializeParams)|initialize|Mapping from Set Token to initialization parameters|
+
+
+#### Functions
+
+| Name  | Caller  | Description 	|
+|------	|------	|-------------	|
+|createSetAndManager |deployer|Create new Set Token with a DelegatedManager manager|
+|createManager | SetToken owner |Migrate existing Set Token to a DelegatedManager manager|
+|initialize |deployer or SetToken owner |Initialize modules and extensions|
+
+> createSetAndManager
+
+```solidity
+function createSetAndManager(
+    address[] memory _components,
+    uint256 _units,
+    string memory _name,
+    string memory _symbol,
+    address _owner,
+    address _methodologist,
+    address[] memory _modules,
+    address[] memory _operators,
+    address[] memory _assets,
+    address[] memory _extensions
+)
+    external
+{
+    if (_assets.length != 0) {
+        _validateComponentsMatchWhitelistedAssets(_components, assets);
+    }
+
+    address setTokenAddress = _deploySet(
+        _components,
+        _modules,
+        _units,
+        _name,
+        _symbol
+    );
+
+    address managerAddress = _deployManager(
+        setTokenAddress,
+        _methodologist,
+        _operators,
+        _assets,
+        _extensions
+    );
+
+    initialize[setTokenAddress] = InitializeParams({
+        deployer: msg.sender,
+        owner: _owner,
+        manager: managerAddress,
+        isPending: true
+    });
```

Comment:
We need to call `setManager` here and pass in the `managerAddress`

Reply:
Have deferred this call so that it happens conditionally (if the factory owns the SetToken) in `initialize` after modules and extensions are initialized.